Lesson_8/wineshop_with_cart.html

```diff
@@ -0,0 +1,128 @@
+        </div>
+    </div>
+    <br>
+    <hr />
+    <div>
+        <h3>Shopping Cart</h3>
+        <button id="showShoppingCartBtn">Show shopping cart</button>
+        <div id="shoppingCart">
+        </div>
+        <h3>Total Price</h3>
+        <div id="total"></div>
+    </div>
+
+    <script>
+        let wines = [
+            {
+                title: 'Primitivo',
+                type: 'red',
+                origin: { country: 'Italy', region: 'Salento', winery: 'Verosso' },
+                year: 2016,
+                grapes: 'primitivo',
+                price: 4.34,
+                currency: '€'
+            },
+            {
+                title: 'Chianti',
+                type: 'red',
+                origin: { country: 'Italy', region: 'Chianti', winery: 'Piccini' },
+                year: 2017,
+                grapes: 'Cabernet Sauvignon, Merlot, Sangiovese, Tempranillo, Ciliegiolo',
+                price: 11.82,
+                currency: '€'
+            },
+            {
+                title: 'Amarone della Valpolicella Classico',
+                type: 'red',
+                origin: { country: 'Italy', region: 'Amarone della Valpolicella Classico', winery: 'Tommasi' },
+                year: 2015,
+                grapes: 'Rondinella, Molinara, Corvinone, Corvina Veronese',
+                price: 35.90,
+                currency: '€'
+            },
+            {
+                title: 'Valdobbiadene Millesimato Brut 2017',
+                type: 'sparkling',
+                origin: { country: 'Italy', region: 'Conegliano-Valdobbiadene Prosecco', winery: 'Borgo Molino' },
+                year: 2019,
+                grapes: 'Glera',
+                price: 14.90,
+                currency: '€'
+            },
+            {
+                title: 'Oberhäuser Brücke Riesling Spätlese 2015',
+                type: 'white',
+                origin: { country: 'Germany', region: 'Nahe', winery: 'Dönnhoff' },
+                year: 2015,
+                grapes: 'Riesling',
+                price: 26.50,
+                currency: '€'
+            },
+        ]
+        let shoppingCart = [];
+
+        for (let i = 0; i < wines.length; i++) {
+            const wineListEl = document.getElementById('wineList');
+            wineListEl.innerHTML = wineListEl.innerHTML + `
+            <p id="wineTitle">${wines[i].title}</p>
+            <ul>
+                <li>Type: ${wines[i].type}</li>
+                <li>Origin: ${wines[i].origin.country}/${wines[i].origin.region}/${wines[i].origin.winery}</li>
+                <li>Year: ${wines[i].year}</li>
+                <li>Grapes: ${wines[i].grapes}</li>
+                <li>Price: ${wines[i].price} ${wines[i].currency}</li>
+            </ul>
+            <button id="addWineBtn${i}">Add to cart</button>
+            `;
+        }
+
+        for (let i = 0; i < wines.length; i++) {
+            const addWineBtn = document.getElementById('addWineBtn' + i);
```

Comment:
Good solution to add event listeners to buttons. Problems would start is you will have huge list of items, because you need to go through wines array twice - first to display an information and then second time to assign handler to a buttons. You can do the same thing in one for loop. 
For that you would need to create buttons with  document.createElement and then assign handler the same way as you did below.
But you don't need to change your code, this is fine for now.

Reply:
Thank you! I think it was my problem before that in one loop it didn't work. Buttons were created but addEventListener worked only at the last button. As far as I understood it was because the next created button deleted previous addEventListeners somehow but I am not sure now.